### PR TITLE
Nuget package changes: Use a meta-package, rename to runtime, remove other DLLs and EXEs

### PR DESCRIPTION
**Pull request by jefgen**
## Summary
This PR makes a number of changes to the Nuget build scripts and configuration for MS-ICU.

It removes the all-in-one Nuget package approach that was used previously, and instead now uses a meta-package with dependencies. These dependent packages contain architecture specific DLLs, which can be built separately from the main meta-package.

The name of the package is changed to "`Microsoft.ICU.icu4c.runtime`" to reflect that it only contains the 'common' and 'i18n' code libraries from ICU4C, along with the data DLL. No other binaries are included in the package. (Ex: the ICU tools or the IO library).
The headers and libs/pdbs are also omitted as well, in order to make the "runtime" package as small as possible.

## PR Checklist
* [x] I have verified that my change is specific to this fork and cannot be made upstream.
* [ ] I am making a maintenance related change.
* [ ] I am making a change that is related to usage internal to Microsoft.
* [ ] I am making a change that is related to the Windows OS build of ICU.
* [x] CLA signed. If not, please see [here](https://cla.opensource.microsoft.com/microsoft/icu) to sign the CLA.

### CODE REVIEW

File: build/nuget/Template-runtime-meta.nuspec [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
     <copyright>© Microsoft Corporation. All rights reserved.</copyright>
     <tags>native nativepackage icu icu4c</tags>
+    <dependencies>
+$deps$
+    </dependencies>
   </metadata>
-</package>
\ No newline at end of file
+</package>
+
```
Comment (daniel-ju) [inline build/nuget/Template-runtime-meta.nuspec:27]: nit: whitespace

File: build/nuget/Template-runtime-meta.nuspec [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
-  <metadata>
-    <id>$id$</id>
+<?xml version="1.0" encoding="utf-8"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2013/01/nuspec.xsd">
+  <metadata minClientVersion="2.12">
+    <id>$runtimePackageId$</id>
     <version>$version$</version>
     <authors>Microsoft</authors>
     <license type="file">LICENSE</license>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
```
Comment (daniel-ju) [inline build/nuget/Template-runtime-meta.nuspec:8]: >false [](start = 30, length = 5)

Why's this false now?

File: build/nuget/Template-runtime-meta.nuspec (hunk 1 above)
Comment (jefgen) [reply build/nuget/Template-runtime-meta.nuspec]: Thanks!

File: build/nuget/Template-runtime-meta.nuspec (hunk 2 above)
Comment (jefgen) [reply build/nuget/Template-runtime-meta.nuspec]: Good point. I put it in the dependent runtime packages, as the meta package should have it, but looks like I also copied it in the meta package too.
Will remove.

Comment (daniel-ju) [reply build/nuget/Template-runtime-meta.nuspec]: Should the runtime package have this set to true?

Comment (jefgen) [reply build/nuget/Template-runtime-meta.nuspec]: Hm, it seems like the only thing that actually does anything with this is the VS installer. Nuget.exe just ignores it...
I suppose it is probably better to set it anyways though...

**Pull request merged**